Assets/Scripts/EntityController.cs

```diff
@@ -18,6 +18,8 @@ public class EntityController : MonoBehaviour
 
     protected Vector2 targetPos;
 
+    public GameEnums.EntityType entityType = GameEnums.EntityType.DISTANCE;
```

Comment:
Instead of having this variable, how about entities having a reference to their data structure?
Aka `public UnitData unitData;`

With this, you can have access to the default data of the unit, which includes their attack range type, without having redundant variables.
The setup of this variable can either be via editor (in case of the enemies) or when the unit is spawned (check the [spawning section](https://github.com/hsven/mystic-militia/blob/d6d70940200fcb5f35c8b5a6add67951d7270672/Assets/Scripts/BattleManager.cs#L60) of the BattleManager script).